.github/workflows/main.yml

```diff
@@ -46,3 +46,11 @@ jobs:
 
     - name: Run tests
       run: bundle exec rake test
+
+    - if: matrix.ruby == 3.1 && matrix.rails == 7
+      name: Report coverage
+      uses: paambaati/codeclimate-action@v3.0.0
```

Comment:
This might be the reason to why your CodeClimate setup is failing. This Github action is not supported by us, it may contain some pitfalls we recommend using our default tools, see https://docs.codeclimate.com/docs/configuring-test-coverage.

Reply:
Alright, I got rid of it. Unfortunately, I still cannot make this run. Same error.

ps: I got that gh action from https://docs.codeclimate.com/docs/github-actions-test-coverage

Reply:
It runs now, after downgrading simplecov to version 0.17.1 as mentioned [here](https://gist.github.com/jcypret/608731eed983dd46a9e7447791c86cdb?permalink_comment_id=4013611#gistcomment-4013611).